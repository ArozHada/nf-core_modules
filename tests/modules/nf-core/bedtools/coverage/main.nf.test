nextflow_process {

    name "Test Process BEDTOOLS_COVERAGE"
    script "modules/nf-core/bedtools/coverage/main.nf"
    process "BEDTOOLS_COVERAGE"
    config "tests/modules/nf-core/bedtools/coverage/nextflow.config"
    tag "bedtools_coverage"

    test("Sarscov2 Genome BED") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test', single_end:false ], // meta map
                            file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
                           ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("coverage") }
            )
        }
    }

    test("Sarscov2 Genome BED With Fasta") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test', single_end:false ], // meta map
                            file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
                           ]
                input[1] = file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("coverage_fasta") }
            )
        }
    }
}
