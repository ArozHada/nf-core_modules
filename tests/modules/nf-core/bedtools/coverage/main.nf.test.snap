{
    "coverage": {
        "content": [
            [
                "test.coverage.bed:md5,3d7e917f97bcb728510b614aa117193b",
                "versions.yml:md5,53d0ba29bf99f699cf0ddd862514e897"
            ]
        ],
        "timestamp": "2023-04-11T19:48:39+0000"
    },
    "coverage_fasta": {
        "content": [
            [
                "test.coverage.bed:md5,3d7e917f97bcb728510b614aa117193b",
                "versions.yml:md5,53d0ba29bf99f699cf0ddd862514e897"
            ]
        ],
        "timestamp": "2023-04-11T19:48:39+0000"
    }
}